src/plugins/walletConnectToDapps/components/modals/SessionProposal.tsx

```diff
@@ -180,29 +181,68 @@ const SessionProposal = forwardRef<SessionProposalRef, WalletConnectSessionModal
       )
     }, [accountIdsByChainId, optionalNamespaces, requiredNamespaces, wallet])
 
-    const approvalNamespaces: SessionTypes.Namespaces = useMemo(
-      () => createApprovalNamespaces(requiredNamespaces, optionalNamespaces, selectedAccountIds),
-      [optionalNamespaces, requiredNamespaces, selectedAccountIds],
+    const handleConnectAccountIds = useCallback(
+      async (_selectedAccountIds: AccountId[]) => {
```

Comment:
Now taking AccountIds in as arg, so we can 
- pass all from select all, we can't really wait for the `selectedAccountIds` state to be reflected, since it would be updated on the next render after toggle... but we do need them on the current one when clicking
- pass `selectedAccountIds` in the regular confirm button, same behaviour as before